enhancements/bpfd-on-ocp.md

```diff
@@ -0,0 +1,424 @@
+
+For Deployment scenario 3, the application would need to have a centralized
+operator. In this case, the operator will use the bpfd-helpers library to manage
+the `bpfProgramConfig` object and dynamically configure fields such as bpfd
+bytecode location, attach points, priority (when applicable), direction, etc.
+
+### API Extensions
+
+The bpfd operator uses the following APIs:
+
+#### BpfProgramConfig CRD
+
+The [BpfProgramConfig
+CRD](https://github.com/redhat-et/bpfd/blob/main/bpfd-operator/config/crd/bases/bpfd.io_bpfprogramconfigs.yaml)
+is the bpfd K8s API object most relevant to users and can be used to understand
+clusterwide state for a BPF program. It's designed to express how, and where BPF
+programs are to be deployed within a Kubernetes cluster. An example
+BpfProgramConfig which loads a basic xdp-pass program to all nodes can be seen
+below:
+
+```yaml
+apiVersion: bpfd.io/v1alpha1
+kind: BpfProgramConfig
+metadata:
+  labels:
+    app.kubernetes.io/name: BpfProgramConfig
+  name: xdp-pass-all-nodes
+spec:
+  ## Must correspond to image section name
+  name: pass
+  type: XDP
+  # Select all nodes
+  nodeselector: {}
+  priority: 0
+  attachpoint: 
+    interface: eth0
+  bytecode:
+    imageurl: quay.io/bpfd-bytecode/xdp_pass:latest
+```
+
+
+#### BpfProgram CRD
+
+The [BpfProgram
+CRD](https://github.com/redhat-et/bpfd/blob/main/bpfd-operator/config/crd/bases/bpfd.io_bpfprograms.yaml)
+is used internally by the bpfd-deployment to keep track of per node bpf program
+state such as program UUIDs and map pin points, and to report node specific
+errors back to the user. K8s users/controllers are only allowed to view these
+objects, NOT create or edit them. Below is an example BpfProgram Object which
+was automatically generated in response to the above BpfProgramConfig Object.
+
+```yaml
+apiVersion: bpfd.io/v1alpha1
+  kind: BpfProgram
+  metadata:
+    creationTimestamp: "2022-12-07T22:41:29Z"
+    finalizers:
+    - bpfd.io.agent/finalizer
+    generation: 2
+    labels:
+      owningConfig: xdp-pass-all-nodes
+    name: xdp-pass-all-nodes-bpfd-deployment-worker2
+    ownerReferences:
+    - apiVersion: bpfd.io/v1alpha1
+      blockOwnerDeletion: true
+      controller: true
+      kind: BpfProgramConfig
+      name: xdp-pass-all-nodes
+      uid: 6e3f5851-97b1-4772-906b-3ac69c6a4057
+    resourceVersion: "1506"
+    uid: 384d3d5c-e62b-4be3-9bf0-c6cf0e315acf
+  spec:
+    programs:
+      bdeac6d3-4128-464e-9161-6010684eca27:
+        attachpoint:
+          interface: eth0
+        maps: {}
+  status:
+    conditions:
+    - lastTransitionTime: "2022-12-07T22:41:30Z"
+      message: Successfully loaded BpfProgram
+      reason: bpfdLoaded
+      status: "True"
+      type: Loaded
+```
+
+#### Bpfd Configmap
+
+The [bpfd
+configmap](https://github.com/redhat-et/bpfd/blob/main/bpfd-operator/bundle/manifests/bpfd-config_v1_configmap.yaml)
+is used to configure cluster-wide parameters.
```

Comment:
Is there also an API to control the contents of that config map, or do admins have to edit it directly?

What sort of parameters does it hold?